### DISCUSSION TITLE
Is logging available?

### DISCUSSION THREAD

**Discussion by matthewmcneely** [SigNoz/signoz#1650] [Q&A]
I've seen https://signoz.io/docs/userguide/logs/#collecting-logs. And all that seems golden. I'm seeing syslogs from the docker containers.

However, in the go sample app, I've:

```
	loggerInstance = stdr.New(log.New(os.Stdout, "", log.LstdFlags|log.Lshortfile))
	otel.SetLogger(loggerInstance)
```

(taken from the opentelemetry site) and then sent some log messages through loggerInstance, but they are not showing up in the signoz dashboard. Any tips?

**Comment by pranay01** [comment]
Yes, @matthewmcneely logging is available in SigNoz

@nityanandagohain do you have more insights on what could be going on here

**Reply by matthewmcneely** [reply]
Thanks @pranay01 

If it's any help, here's my docker environment:
```
CONTAINER ID   IMAGE                                        COMMAND                  CREATED       STATUS                 PORTS                              NAMES
d77e32755a75   signoz/frontend:0.11.2                       "nginx -g 'daemon of…"   3 hours ago   Up 2 hours             80/tcp, 0.0.0.0:3301->3301/tcp     frontend
b8fa599d1519   signoz/alertmanager:0.23.0-0.2               "/bin/alertmanager -…"   3 hours ago   Up 2 hours             9093/tcp                           clickhouse-setup_alertmanager_1
20018d210b0f   signoz/signoz-otel-collector:0.55.3          "/signoz-collector -…"   3 hours ago   Up 2 hours             0.0.0.0:4317-4318->4317-4318/tcp   clickhouse-setup_otel-collector_1
d55355c879ce   signoz/query-service:0.11.2                  "./query-service -co…"   3 hours ago   Up 2 hours (healthy)   8080/tcp                           query-service
3f2717638777   signoz/signoz-otel-collector:0.55.3          "/signoz-collector -…"   3 hours ago   Up 2 hours             4317-4318/tcp                      clickhouse-setup_otel-collector-metrics_1
e43f5ea94cc9   clickhouse/clickhouse-server:22.4.5-alpine   "/entrypoint.sh"         3 hours ago   Up 2 hours (healthy)   8123/tcp, 9000/tcp, 9009/tcp       clickhouse-setup_clickhouse_1
```

**Reply by matthewmcneely** [reply]
Also, here's the log call that I added in the [sample app](https://github.com/SigNoz/sample-golang-app) in one of the handlers:
```
loggerInstance.Info("getting all books")
```

**Comment by nityanandagohain** [comment]
Hi @matthewmcneely  for debugging this can you check a few things?
* Check if your application logs are getting printed by tailing the logs `docker logs --tail 100 <container_name>`
* Check the logs of the signoz_otel_collector, if there is any log that says no file found inside `/var/lib/docker/containers/`

**Reply by matthewmcneely** [reply]
> * Check if your application logs are getting printed by tailing the logs `docker logs --tail 100 <container_name>`

I'm using your sample-golang-app to test the log, so it's not in a container (just running `go run main.go`).... logs are not getting written to the console.

I checked the logs for `clickhouse-setup-otel-collector-1` (which I think is the right container) and I see no message pertaining to /var/lib/docker/containers

The last entry in that log might be of interest however:
```
2022-11-08T16:46:30.569Z	info	exporterhelper/queued_retry.go:215	Exporting failed. Will retry the request after interval.	{"kind": "exporter", "data_type": "logs", "name": "clickhouselogsexporter", "error": "PrepareBatch:read tcp 172.28.0.4:49990->172.28.0.2:9000: i/o timeout", "interval": "5.52330144s"}
```

This is invoked when I touch a REST endpoint of the sample-golang-app in which I added a `loggerInstance.Info(...)` call.
